middle_end/flambda2/simplify/simplify_let_cont_expr.ml

```diff
@@ -1162,36 +1257,45 @@ let simplify_recursive_handlers ~rebuild_body ~invariant_params ~invariant_epa
   in
   loop
 
-let after_downwards_traversal_of_body ~simplify_expr
-    (data : after_downwards_traversal_of_body_data) ~down_to_up dacc
-    ~rebuild:rebuild_body =
-  (* At this point, we have done the downwards traversal of the body. We prepare
-     to loop over all the handlers defined by the let cont. *)
+and simplify_handlers ~simplify_expr ~down_to_up ~denv_for_join ~rebuild_body
+    (data : after_downwards_traversal_of_body_data) dacc =
+  (* In this case we have decided not to lift the continuation being let-bound
+     outside of its context. So the remaining thing to do is setup the dacc, and
+     later decide whether to lift the continuations defined inside the
+     continuation being currently let-bound. *)
   let body_continuation_uses_env = DA.continuation_uses_env dacc in
-  let denv = data.denv_before_body in
+  let denv = data.denv_for_join in
   let consts_lifted_during_body = DA.get_lifted_constants dacc in
   let dacc =
     DA.add_to_lifted_constant_accumulator dacc data.prior_lifted_constants
   in
+  let previous_are_lifting_conts = DA.are_lifting_conts dacc in
   match data.handlers with
-  | Non_recursive { cont; params; handler; is_exn_handler; is_cold } -> (
+  | Non_recursive
+      { cont; params; lifted_params; handler; is_exn_handler; is_cold } -> (
     match
       Continuation_uses_env.get_continuation_uses body_continuation_uses_env
         cont
     with
     | None ->
       (* Continuation unused, no need to traverse its handler *)
-      let (data : after_downwards_traversal_of_body_and_handlers_data) =
+      let data : after_downwards_traversal_of_body_and_handlers_data =
         { rebuild_body;
           cont_uses_env = body_continuation_uses_env;
+          lifted_params = Lifted_cont_params.empty;
           invariant_params = Bound_parameters.empty;
           invariant_extra_params_and_args = EPA.empty;
           handlers = Continuation.Map.empty;
           at_unit_toplevel = false
         }
       in
-      after_downwards_traversal_of_body_and_handlers data ~down_to_up dacc
+      after_downwards_traversal_of_body_and_handlers data dacc
+        ~simplify_expr ~denv_for_join ~down_to_up
```

Comment:
I'm wondering whether we should include `denv_for_join` in `data` here (I don't have a clear answer though).